src/datasource.js

```diff
@@ -74,31 +89,35 @@ export class HawkularDatasource {
   }
 
   testDatasource() {
+    // If tenants is unknown at this point (when having per-query tenants)
+    // We do a more basic check to status endpoint
+    // Else, it's full connectivity with tenant check
+    const endpoint = this.isTenantPerQuery ? '/status' : '/metrics';
```

Comment:
You're right, but I can't see other way round. In openshift for example authentication is closely linked to the tenant, so it cannot be checked anyway at this step.